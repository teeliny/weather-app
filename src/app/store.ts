import { configureStore } from '@reduxjs/toolkit';

import {  apiSlice } from '../features/weather/weather-api-slice';
import screenReducer from '../features/screen/screen-width-slice';

export const store = configureStore({
  reducer: {
    screen: screenReducer,
    [apiSlice.reducerPath]: apiSlice.reducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware().concat(apiSlice.middleware);
  }
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
