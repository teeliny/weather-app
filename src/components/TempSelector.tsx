import { FormControl, RadioGroup, FormControlLabel, Radio } from '@mui/material';
import React from 'react';

interface ITempSelector {
  value: string;
  handleChange: (e: React.ChangeEvent<HTMLInputElement>, value: string) => void;
}

function TempSelector({value, handleChange}: ITempSelector) {
  return (
    <>
      <FormControl component="fieldset" sx={{ width: '100%' }}>
        <RadioGroup
          sx={{ 
            width: '100%', 
            display: 'flex', 
            justifyContent: 'space-between',
          }}
          row={true}
          aria-label="temp"
          name="row-radio-buttons-group"
          value={value}
          onChange={handleChange}
        >
          <FormControlLabel
            value="0"
            control={<Radio />}
            label="Celsius"
          />
          <FormControlLabel
            value="1"
            control={<Radio />}
            label="Fahrenheit"
            sx={{ margin: 0 }}
          />
        </RadioGroup>
      </FormControl>
    </>
  )
}

export default TempSelector;
