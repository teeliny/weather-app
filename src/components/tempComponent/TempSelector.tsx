import FormControl from '@mui/material/FormControl';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Radio from '@mui/material/Radio';
// import Button from '@mui/material/Button';
import React from 'react';
import { useAppSelector } from '../../app/hooks';
import { TempWrapper } from './temp.style';
import { ITempSelector } from './temp.typing';


function TempSelector({ value, handleChange, handleRefetch }: ITempSelector) {
  const myView = useAppSelector((state) => state.screen.size_view);

  return (
    <TempWrapper data-testid={'temp-control'}>
      <FormControl component="fieldset" sx={{ width: '100%' }}>
        <RadioGroup
          sx={{
            width: '100%',
            display: 'flex',
            justifyContent: myView ? 'space-between' : 'center',
          }}
          row={true}
          aria-label="temp"
          name="row-radio-buttons-group"
          value={value}
          onChange={handleChange}
        >
          <FormControlLabel value="0" control={<Radio />} label="Celsius" />
          <FormControlLabel
            value="1"
            control={<Radio />}
            label="Fahrenheit"
            sx={{ margin: 0 }}
          />
        </RadioGroup>
        {/* {!myView && (
          <Button
            sx={{
              position: 'absolute',
              right: '0',
              textTransform: 'capitalize',
              backgroundColor: '#ffffff',
              border: '1px solid #0055ff',
              color: '#0055ff',
              width: '80px',
            }}
            onClick={handleRefetch}
          >
            Refresh
          </Button>
        )} */}
      </FormControl>
    </TempWrapper>
  );
}

export default TempSelector;
