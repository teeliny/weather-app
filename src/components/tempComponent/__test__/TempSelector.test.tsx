import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import TempSelector from '../TempSelector';

let eleWrapper: JSX.Element;

beforeEach(() => {
  eleWrapper = (
    <Provider store={store}>
      <TempSelector
        value={''}
        handleChange={function (
          e: React.ChangeEvent<HTMLInputElement>,
          value: string,
        ): void {
          throw new Error('Function not implemented.');
        }}
        handleRefetch={function (): void {
          throw new Error('Function not implemented.');
        }}
      />
    </Provider>
  );
});
afterEach(cleanup);

it('render temp selector correctly', () => {
  const { getByTestId } = render(eleWrapper);
  expect(getByTestId('temp-control')).toHaveTextContent('Fahrenheit');
  expect(getByTestId('temp-control')).toHaveTextContent('Celsius');
});

it('matches temp selector snapshot', () => {
  const tree = renderer.create(eleWrapper).toJSON();
  expect(tree).toMatchSnapshot();
})