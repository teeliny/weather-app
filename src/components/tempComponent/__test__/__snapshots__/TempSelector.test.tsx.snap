// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches temp selector snapshot 1`] = `
<div
  className="css-14530gt"
  data-testid="temp-control"
>
  <fieldset
    className="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
  >
    <div
      aria-label="temp"
      className="MuiFormGroup-root MuiFormGroup-row css-dyh9s9-MuiFormGroup-root"
      role="radiogroup"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="row-radio-buttons-group"
            onChange={[Function]}
            type="radio"
            value="0"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Celsius
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-fcbanr-MuiFormControlLabel-root"
      >
        <span
          className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            disabled={false}
            name="row-radio-buttons-group"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Fahrenheit
        </span>
      </label>
    </div>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-172vse9-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Refresh
    </button>
  </fieldset>
</div>
`;
